error_middleware/error_handler.js

```diff
@@ -0,0 +1,11 @@
+'use strict';
+
+module.exports = function AppError(message, httpStatus = 404) {
```

Comment:
AppError looks really meaningless and httpStatus = 404 assuming httpStatus as 404 is wrong. 

In error_middleware, 

Add various files  like,
internal-server-error.js - default HTTP  code - 500
not-found-error.js -  default HTTP  code - 404
conflict-error.js  - default HTTP  code - 409